evm_arithmetization/src/generation/state.rs

```diff
@@ -350,13 +350,16 @@ impl<F: Field> GenerationState<F> {
     /// we're jumping to a special location.
     pub(crate) fn jump_to(&mut self, dst: usize) -> Result<(), ProgramError> {
         self.registers.program_counter = dst;
-        if dst == KERNEL.global_labels["observe_new_address"] {
-            let tip_u256 = stack_peek(self, 0)?;
+        if dst == KERNEL.global_labels["observe_new_address"] && self.get_registers().stack_len > 0
```

Comment:
minor, but I'd just return early if stack is empty:

```rust
if self.stack().is_empty() {
    // We cannot observe anything as the stack is empty.
    return Ok(());
}
```
and leave what's below unchanged